src/ResourceManager/Sql/Commands.Sql/Common/AzureSqlCmdletBaseBase.cs

```diff
@@ -0,0 +1,169 @@
+                {
+                    if (!string.IsNullOrEmpty(databaseName))
+                    {
+                        return string.Format("{0}.{1}", serverName, databaseName);
+                    }
+                    return serverName;
+                }
+                if (!string.IsNullOrEmpty(databaseName))
+                {
+                    return databaseName;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// The ModelAdapter object used by this cmdlet
+        /// </summary>
+        public A ModelAdapter { get; internal set; }
+
+        /// <summary>
+        /// Gets an entity from the service
+        /// </summary>
+        /// <returns>A model object</returns>
+        protected abstract M GetEntity();
+
+        /// <summary>
+        /// Updates the given model element with the cmdlet specific operation 
+        /// </summary>
+        /// <param name="model">A model object</param>
+        protected virtual M ApplyUserInputToModel(M model) { return model; }
+
+        /// <summary>
+        /// This method is responsible to call the right API in the communication layer that will eventually send the information in the 
+        /// object to the REST endpoint
+        /// </summary>
+        /// <param name="entity">The model object with the data to be sent to the REST endpoints</param>
+        protected virtual M PersistChanges(M entity) { return default(M); }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected virtual bool WriteResult() { return true; }
+
+        /// <summary>
+        /// Creation and initialization of the ModelAdapter object
+        /// </summary>
+        /// <param name="subscription">The AzureSubscription in which the current execution is performed</param>
+        /// <returns>An initialized and ready to use ModelAdapter object</returns>
+        protected abstract A InitModelAdapter(IAzureSubscription subscription);
+
+        /// <summary>
+        /// Transforms the given model object to be an object that is written out
+        /// </summary>
+        /// <param name="model">The about to be written model object</param>
+        /// <returns>The prepared object to be written out</returns>
+        protected virtual object TransformModelToOutputObject(M model)
+        {
+            return model;
+        }
+
+        protected virtual string GetConfirmActionProcessMessage()
+        {
+            return Properties.Resources.BaseConfirmActionProcessMessage;
+        }
+
+        /// <summary>
+        /// Executes the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            ModelAdapter = InitModelAdapter(DefaultProfile.DefaultContext.Subscription);
```

Comment:
@shahabhijeet InitModelAdapter is implemented in each individual cmdlet. [Database Cmdlet base example](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/AzureSqlDatabaseCmdletBase.cs#L42)